### PR TITLE
Added Simple Upload Adapter

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Introduced `SimpleUploadAdapter` plugin which enables file uploads in CKEditor 5 using the external side-server connection. Closes ckeditor/ckeditor5#1791.

BREAKING CHANGE: Moved the `Base64UploadAdapter` plugin to `ckeditor5-upload/src/adapters/base64uploadadapter.js`.

### CODE REVIEW

**Comment by oleq** [comment]
What do we do about https://ckeditor.com/docs/ckeditor5/latest/framework/guides/deep-dive/upload-adapter.html in this situation? It feels like it needs a solid refactoring.

cc @mlewand @Reinmar

File: src/simpleuploadadapter.js [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+	/**
+	 * Initializes the XMLHttpRequest object using the URL passed to the constructor.
+	 *
+	 * @private
+	 */
+	_initRequest() {
+		const xhr = this.xhr = new XMLHttpRequest();
+
+		xhr.open( 'POST', this.options.uploadUrl, true );
```
Comment (oleq) [inline src/simpleuploadadapter.js]: We must clearly document that the plugin will make a POST request.

**Comment by pomek** [comment]
@oleq, could you review it again? I extended API page and the SImple upload adapter guide. I guess, all your notes have been introduced.

File: docs/features/simple-upload-adapter.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		plugins: [ SimpleUploadAdapter, ... ],
+		toolbar: [ ... ],
+		simpleUpload: {
+			...
+		}
+	} )
+	.then( ... )
+	.catch( ... );
+```
+
```
[the remark this answers is not in the document]
Comment (pomek) [reply docs/features/simple-upload-adapter.md]: I am not sure what more we could do here.

**Comment by pomek** [comment]
If this PR won't be merged in the current iteration (_i25_), changes in this PR are BC.

**Pull request merged**